### PR TITLE
Feature/realtime notifications

### PR DESCRIPTION
**Pull request by Somiel**
Implement real-time notifications for drivers and managers to track a newcoming orders

### CODE REVIEW

File: app/controllers/notifications_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Notifications controller
+class NotificationsController < FayeRails::Controller
+  observe Order, :after_create do |new_order|
```
Comment (murderface13) [inline app/controllers/notifications_controller.rb]: why you didn't use rails observers?
http://api.rubyonrails.org/v3.2.0/classes/ActiveRecord/Observer.html

File: app/views/layouts/application.html.slim [hunk 2]
```diff
@@ -6,6 +6,9 @@ html
     = stylesheet_link_tag    'home', media: 'all', 'data-turbolinks-track' => true
     = stylesheet_link_tag    'orders', media: 'all', 'data-turbolinks-track' => true
     = javascript_include_tag 'application', 'data-turbolinks-track' => true
+
+
+    = javascript_include_tag root_url + "faye/client.js"
```
Comment (murderface13) [inline app/views/layouts/application.html.slim]: why you include one more js file in layot? it is good idea to use asset pipeline, i think. isn't it?

File: app/assets/javascripts/orders.js.coffee [hunk 3]
```diff
@@ -5,6 +5,84 @@ $(document).on 'ready page:change', (e) ->
...
+$(document).on 'ready', ->
+  if $('table').is('.orders-list')
+    client = new Faye.Client( '/faye', {
+      timeout: 120
+    })
+
+    channel = "notifications"
+
+    subscription = client.subscribe('/' + channel, (message) ->
+      if message[0].type == 'create'
```
Comment (murderface13) [inline app/assets/javascripts/orders.js.coffee]: what about switch-case operator?

Comment (Somiel) [inline app/assets/javascripts/orders.js.coffee]: There are only 3 cases here..but i can use switch-case, thats np

File: app/views/layouts/application.html.slim (hunk 2 above)
Comment (Somiel) [inline app/views/layouts/application.html.slim]: do you wanna me to include this js file in index template?

File: app/controllers/notifications_controller.rb (hunk 1 above)
Comment (Somiel) [inline app/controllers/notifications_controller.rb]: I`ve seen it in 'faye-rails' docs..ill think about changing an existing solution

File: app/assets/javascripts/orders.js.coffee (hunk 3 above)
Comment (Somiel) [inline app/assets/javascripts/orders.js.coffee]: refactored

File: app/views/layouts/application.html.slim (hunk 2 above)
Comment (Somiel) [inline app/views/layouts/application.html.slim]: removed

File: app/controllers/notifications_controller.rb (hunk 1 above)
Comment (Somiel) [inline app/controllers/notifications_controller.rb]: issue closed :smile:

**Comment by murderface13** [comment]
Looks good :+1:

**Pull request merged**